### PR TITLE
#2147 - Implemented the Data Export Option for Lighthouse SPA

### PR DESCRIPTION
**Pull request by riginoommen**
# Resolves 2147

# Explain the feature

Implemented Data Export Option for Lighthouse SPA and Microservice

## Does this PR introduce a breaking change

No

## Screenshots
<details>
- Created button in UI for csv download

<summary>View Screenshots</summary>

image
</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demoed and the design review approved?

### CODE REVIEW

**Review by deshmukhmayur — APPROVED** [review]

File: packages/lighthouse-spa/src/app/dashboard/pages/analysis/analysis.component.ts [hunk 1]
```diff
@@ -255,4 +256,10 @@ export class AnalysisComponent implements OnInit {
     );
     this.fetchLHLeaderboard();
   }
+
+  exportLHReportByProject () {
+    this.dashboardService.exportLHReportByProject( this.projectId, this.selectedBranch ).subscribe( data => {
```
Comment (akhilmhdh) [inline packages/lighthouse-spa/src/app/dashboard/pages/analysis/analysis.component.ts:261]: Remove the subscription on unmount by passing the destroying sub

**Review by akhilmhdh — CHANGES_REQUESTED** [review]
Kindly write the testcase for this LH functionality

**Review by deshmukhmayur — APPROVED** [review]
@riginoommen Is this ready for merging?

**Review by hybridx — APPROVED** [review]
LGTM

**Pull request merged**